llama_stack/providers/remote/inference/groq/config.py

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from typing import Optional
+
+from llama_models.schema_utils import json_schema_type
+from pydantic import BaseModel, Field
+
+
+@json_schema_type
+class GroqConfig(BaseModel):
+    api_key: Optional[str] = Field(
+        default=None,
```

Comment:
nit - the groq library will read GROQ_API_KEY env (https://github.com/groq/groq-python/blob/main/src/groq/_client.py#L86), consider adding a comment here so people in the LS codebase know this expectation

Reply:
Done

Reply:
Let's not rely on environment variables for code that we expect to run in llama-stack server. We would want to take in the api key as a config variable in run.yaml when we spin up the server

Reply:
> We would want to take in the api key as a config variable in run.yaml when we spin up the server

@raghotham, I believe that's the behaviour at the moment. This is how fireworks and together define their configs:

https://github.com/meta-llama/llama-stack/blob/96e158eaac4aca62a62afeae40558e053627e547/llama_stack/providers/remote/inference/fireworks/config.py#L19-L22

https://github.com/meta-llama/llama-stack/blob/6765fd76fff516e654390ab9b21d74b6299ebd29/llama_stack/providers/remote/inference/together/config.py#L19-L22

And it's in the run.yaml that you can template in the environment variable:


wdyt?

Reply:
yes @aidando73 this is correct. Note that both Together and Fireworks also support grabbing the api key from headers via the `NeedsProviderData` mixin. You can add that if you feel like it.

Reply:
Done - added the mixin